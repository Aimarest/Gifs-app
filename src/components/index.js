//Esto es un archivo de barril para poder exportar todo a la vez.

export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';