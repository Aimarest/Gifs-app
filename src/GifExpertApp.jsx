import React, { useState } from 'react';
import { AddCategory, GifGrid } from './components';


export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['FRIENDS', 'BAILE', 'ANIMALS']);

    const handleAddCategory = (newCategory) => {

        if(categories.includes(newCategory.toUpperCase())) return;

        setCategories([newCategory.toUpperCase(),...categories ])
    }
    return (
        <>

            <header className='header'>
                <h1> GifExpertApp </h1>
                <AddCategory handleAddCategory={handleAddCategory} />
            </header>
            {
                categories.map((category) => (

                    <GifGrid category={category} key={category} />

                )

                )
            }

        </>
    );
}

