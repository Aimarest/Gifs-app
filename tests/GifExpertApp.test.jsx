
import { render, screen, fireEvent } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Pruebas sobre el componente GifExpertApp', () => {
    

    test('Debe hacer match con el snapshot', () => { 

       const { container } = render( <GifExpertApp />)
     
        expect( container ).toMatchSnapshot();
    })

    test('Debe tener un titulo que ponga GifExpertApp', () => { 
        render( <GifExpertApp />)
       expect( screen.getByText('GifExpertApp') ); 

     })

     test('En la app no debe dejar añadir  categorías repetidas', () => { 
         
        const inputValue = 'FRIENDS';

         render(<GifExpertApp />)

         const input = screen.getByRole('textbox');
         const form = screen.getByRole('form');

         fireEvent.input(input, { target: { value: inputValue } });
         
         fireEvent.submit(form);

         expect(screen.getAllByText(inputValue).length).toBe(1);

     })

     test('Cuando escribo una categoría en el formulario debe aparecer una categoría llamada así', () => { 
        const inputValue = 'KOALA';
        render(<GifExpertApp />)
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target: { value: inputValue }});
        fireEvent.submit(form);

        expect(screen.getByText(inputValue)).toBeTruthy();
      })
});
