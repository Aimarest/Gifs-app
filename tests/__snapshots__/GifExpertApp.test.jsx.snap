// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente GifExpertApp Debe hacer match con el snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <h1>
       GifExpertApp 
    </h1>
    <form
      aria-label="form"
    >
      <input
        placeholder="Buscar gifs"
        type="text"
        value=""
      />
      <button>
         Add a new category 
      </button>
    </form>
  </header>
  <h3>
    FRIENDS
  </h3>
  <h2>
    Cargando imágenes...
  </h2>
  <ul
    class="card-grid"
  />
  <h3>
    BAILE
  </h3>
  <h2>
    Cargando imágenes...
  </h2>
  <ul
    class="card-grid"
  />
  <h3>
    ANIMALS
  </h3>
  <h2>
    Cargando imágenes...
  </h2>
  <ul
    class="card-grid"
  />
</div>
`;
